examples/src/main/kotlin/ai/koog/agents/example/snapshot/FilePersistentAgentExample.kt

```diff
@@ -0,0 +1,129 @@
+package ai.koog.agents.example.snapshot
+
+import ai.koog.agents.core.agent.AIAgent
+import ai.koog.agents.core.agent.config.AIAgentConfig
+import ai.koog.agents.core.tools.ToolRegistry
+import ai.koog.agents.ext.tool.AskUser
+import ai.koog.agents.ext.tool.SayToUser
+import ai.koog.agents.snapshot.feature.AgentCheckpoint
+import ai.koog.agents.snapshot.providers.file.JVMFileAgentCheckpointStorageProvider
+import ai.koog.prompt.dsl.prompt
+import ai.koog.prompt.executor.llms.all.simpleOllamaAIExecutor
+import ai.koog.prompt.executor.model.PromptExecutor
+import ai.koog.prompt.llm.OllamaModels
+import kotlinx.coroutines.runBlocking
+import java.nio.file.Files
+import kotlin.io.path.exists
+import kotlin.uuid.ExperimentalUuidApi
+
+/**
+ * This example demonstrates how to use the file-based checkpoint provider with a persistent agent.
+ * 
+ * The JVMFileAgentCheckpointStorageProvider stores agent checkpoints in a file system,
+ * allowing the agent's state to persist across multiple runs.
+ * 
+ * This example shows:
+ * 1. How to create a file-based checkpoint provider
+ * 2. How to configure an agent with the file-based checkpoint provider
+ * 3. How to run an agent that automatically creates checkpoints
+ * 4. How to restore an agent from a checkpoint
+ */
+@OptIn(ExperimentalUuidApi::class)
+fun main() = runBlocking {
+    // Create a prompt executor for the agent
+    val executor: PromptExecutor = simpleOllamaAIExecutor()
+
+    // Create a temporary directory for storing checkpoints
+    val checkpointDir = Files.createTempDirectory("agent-checkpoints")
+    println("Checkpoint directory: $checkpointDir")
+    
+    // Create the file-based checkpoint provider
+    val provider = JVMFileAgentCheckpointStorageProvider(checkpointDir)
+    
+    // Create a unique agent ID to identify this agent's checkpoints
+    val agentId = "persistent-agent-example"
+    
+    // Create tool registry with basic tools
+    val toolRegistry = ToolRegistry {
+        tool(AskUser)
+        tool(SayToUser)
+    }
+
+    // Create agent config with a system prompt
+    val agentConfig = AIAgentConfig(
+        prompt = prompt("persistent-agent") {
+            system("You are a helpful assistant that remembers conversations across sessions.")
+        },
+        model = OllamaModels.Meta.LLAMA_3_2,
+        maxAgentIterations = 20
+    )
+
+    println("Creating and running agent with continuous persistence...")
+    
+    // Create the agent with the file-based checkpoint provider and continuous persistence
+    val agent = AIAgent(
+        promptExecutor = executor,
+        strategy = SnapshotStrategy.strategy,
```

Comment:
Maybe let's show this example with any existing normal strategy? Like single-run.

The point is to show that now you can take any agent and make it persistent automatically, and deploy to Cloud